package/android/cpp/rnskia-android/SkiaOpenGLSurfaceFactory.cpp

```diff
@@ -0,0 +1,234 @@
+    }
+    return true;
+  }
+  return true;
+}
+
+bool BaseSkiaSurfaceFactory::initializeOpenGL(SurfaceFactoryContext *context) {
+  // Get default display
+  context->glDisplay = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+  if (context->glDisplay == EGL_NO_DISPLAY) {
+    RNSkLogger::logToConsole("eglGetDisplay failed : %i", glGetError());
+    return false;
+  }
+
+  EGLint major;
+  EGLint minor;
+
+  if (eglInitialize(context->glDisplay, &major, &minor) != EGL_TRUE) {
+    RNSkLogger::logToConsole("eglInitialize failed : %i", glGetError());
+    eglTerminate(context->glDisplay);
+    context->glDisplay = EGL_NO_DISPLAY;
+    return false;
+  }
+
+  return true;
+}
+
+EGLConfig BaseSkiaSurfaceFactory::getConfig(EGLDisplay glDisplay) {
+
+  EGLint att[] = {EGL_RENDERABLE_TYPE,
+                  EGL_OPENGL_ES2_BIT,
+                  EGL_ALPHA_SIZE,
+                  8,
+                  EGL_BLUE_SIZE,
+                  8,
+                  EGL_GREEN_SIZE,
+                  8,
+                  EGL_RED_SIZE,
+                  8,
+                  EGL_DEPTH_SIZE,
+                  0,
+                  EGL_STENCIL_SIZE,
+                  0,
+                  EGL_SAMPLE_BUFFERS,
+                  0,
+                  EGL_NONE};
+
+  EGLint numConfigs;
+  EGLConfig glConfig = 0;
+  if (eglChooseConfig(glDisplay, att, &glConfig, 1, &numConfigs) != EGL_TRUE ||
+      numConfigs == 0) {
+    RNSkLogger::logToConsole(
+        "Failed to choose a config for %s surface. Error code: %d\n",
+        eglGetError());
+    return 0;
+  }
+
+  return glConfig;
+}
+
+EGLContext BaseSkiaSurfaceFactory::createOpenGLContext(
+    EGLDisplay glDisplay, EGLConfig config, EGLContext sharedContext) {
+  // Create OpenGL context
+  EGLint contextAttribs[] = {EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE};
+
+  // Initialize the offscreen context for this thread
+  EGLContext newGLContext =
+      eglCreateContext(glDisplay, config, sharedContext, contextAttribs);
+
+  if (newGLContext == EGL_NO_CONTEXT) {
+    RNSkLogger::logToConsole("eglCreateContext failed: %d\n", eglGetError());
+    return EGL_NO_CONTEXT;
+  }
+
+  return newGLContext;
+}
+
+sk_sp<SkSurface> BaseSkiaSurfaceFactory::createSkSurface() {
+  // Get the current context to work in:
+  auto context = getContext();
+
+  // Initialize OpenGL if necessary
+  if (context->glDisplay == EGL_NO_DISPLAY) {
+    if (!initializeOpenGL(context)) {
+      return nullptr;
+    }
+  }
+
+  // Create the OpenGL context if necessary
+  if (context->glContext == EGL_NO_CONTEXT) {
+    // Create config for the current display
+    context->glConfig = getConfig(context->glDisplay);
+    context->glContext =
+        createOpenGLContext(context->glDisplay, context->glConfig,
+                            SharedContext->getContext()->glContext);
+
+    if (context->glContext == EGL_NO_CONTEXT) {
+      return nullptr;
+    }
+  }
+
+  // Create the desired OpenGL surface type
+  _glSurface = createOpenGLSurface(context);
```

Comment:
This seems to suggest that an OpenGL surface is created for every Skia surface (even offscreen). By using the texture API from Skia, making current the surface from which the Skia context was created is enough for the offscreen surface creation. Technically it's only one 1x1 offscreen gl surface per thread.